homeassistant/components/history_stats/data.py

```diff
@@ -186,8 +190,13 @@ def _async_compute_seconds_and_changes(
             current_state_matches = history_state.state in self._entity_states
             state_change_timestamp = history_state.last_changed
```

Comment:
Shouldn't you do the flooring here? or else, you will calculate with the not-floored version moving forward ? (ie elapsed)

Reply:
I did actually try that but it breaks three tests that calculate the time with microsecond precision. 

I'm not sure why that's important, but I don't think there's anything wrong with calculating elapsed with the non-floored time?

```
FAILED tests/components/history_stats/test_sensor.py::test_end_time_with_microseconds_zeroed[Europe/Berlin] - AssertionError: assert '16.0' == '16.0002388888929'
FAILED tests/components/history_stats/test_sensor.py::test_end_time_with_microseconds_zeroed[America/Chicago] - AssertionError: assert '16.0' == '16.0002388888929'
FAILED tests/components/history_stats/test_sensor.py::test_end_time_with_microseconds_zeroed[US/Hawaii] - AssertionError: assert '16.0' == '16.0002388888929'
```

But if you think we should change that I guess we can change the tests also, but seems more disruptive.

Reply:
It was a drive by comment, checking if you knew why. I didn't verify it myself.

Reply:
The fix looks fine 👍